src/pages/index.js

```diff
@@ -0,0 +1,372 @@
+      editFormElement.reset();
+      closeModal(editModal);
+
+      api.getAppInfo()
+        .then(([cards, userInfo]) => {
+          profileName.textContent = userInfo.name;
+          profileDescription.textContent = userInfo.about;
+        })
+        .catch(console.error);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(profileSubmitBtn, false, "Save");
+    });
+}
+
+function toggleSubmitButton() {
+  const isInputValid = cardNameInput.value.trim() !== "" && cardLinkInput.value.trim() !== "";
+  cardSubmitBtn.disabled = !isInputValid; 
+}
+
+cardNameInput.addEventListener("input", toggleSubmitButton);
+cardLinkInput.addEventListener("input", toggleSubmitButton);
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+  cardSubmitBtn.disabled = true; 
+  setButtonText(cardSubmitBtn, true, "Save", "Saving...");
+
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+
+  api.createCard(inputValues)
+    .then((data) => {
+      const cardElement = getCardElement(data);
+      cardsList.prepend(cardElement);
+      cardForm.reset(); 
+      closeModal(cardModal); 
+      cardSubmitBtn.disabled = true; 
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(cardSubmitBtn, false, "Save"); 
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  setButtonText(avatarSubmitBtn, true, "Save", "Saving...");
+
+  api.editAvatarInfo(avatarInput.value)  
+    .then((data) => {
+      avatarImage.src = data.avatar;
+      avatarForm.reset();
+      closeModal(avatarModal);
+
+      api.getAppInfo()
+        .then(([cards, userInfo]) => {
+          avatarImage.src = userInfo.avatar;  
+        })
+        .catch(console.error);  
+    })
+    .catch(console.error) 
+    .finally(() => {
+      setButtonText(avatarSubmitBtn, false, "Save");  
+    });
+}
+
+function closeOnOverlayClick(modal) {
+  modal.addEventListener("click", (evt) => {
+    if (evt.target === modal) {
+      closeModal(modal);
+    }
+  });
+}
+modals.forEach((modal) => {
+  closeOnOverlayClick(modal);
+});
+
+function handleEscape(evt) {
+  if (evt.key === "Escape") {
+    modals.forEach(closeModal);
+  }
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscape);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscape);
+}
+
+profileEditButton.addEventListener("click", () => {
+  editModalNameInput.value = profileName.textContent;
+  resetValidation(
+    editModal,
+    [editModalNameInput],
+    settings
+  );
+  openModal(editModal);
+});
+
+editModalCloseBtn.addEventListener("click", () => {
+  closeModal(editModal);
+});
+
+cardModalBtn.addEventListener("click", () => {
+  cardForm.reset();
+  cardSubmitBtn.disabled = true; 
+  cardSubmitBtn.classList.add("modal__submit-btn_disabled"); 
```

Comment:
You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.